### PR TITLE
[framework] Fix spurious docs changes

### PR DESCRIPTION
**Pull request by amnn**
## Description

Some folks were seeing docs file changes when building `sui-framework` because of residual `build` directories in their repo.  The output produced in CI was *also* broken, but for a different reason -- it couldn't find the location of docs for dependencies.

This change fixes both of those issues by building all system packages in the same "install_dir" (the root of the `sui-framework` crate), and building `sui-framework` before `sui-system`.

## Test Plan

Remove all `build` directories in `sui-frameworks`, run:

```
crates/sui-framework$ touch build.rs && cargo build
```

Then make sure docs don't change.

### CODE REVIEW

**Comment by amanjui45** [comment]
contract deployments should be added

**Comment by amnn** [comment]
(Note that generally docs generation is going to need some love, this change is just to make sure spurious changes don't get added to people's working directories when they build).

File: crates/sui-framework/build.rs [hunk 1]
```diff
@@ -100,19 +102,19 @@ fn build_packages_with_move_config(
     stdlib_dir: &str,
     config: MoveBuildConfig,
 ) {
-    let system_pkg = BuildConfig {
+    let framework_pkg = BuildConfig {
```
Comment (emmazzz) [inline crates/sui-framework/build.rs:105]: Such an elegant fix! Love it ❤️

**Review by emmazzz — APPROVED** [review]
LGTM. Thank you!

**Pull request merged**